addons/point_of_sale/models/pos_payment_method.py

```diff
@@ -97,3 +102,63 @@ def write(self, vals):
     def copy(self, default=None):
         default = dict(default or {}, config_ids=[(5, 0, 0)])
         return super().copy(default)
+
+    @api.constrains('use_qr')
+    def _check_qr_fields(self):
+        for rec in self:
+            if not rec.company_id.qr_code and rec.use_qr:
+                raise ValidationError(_("Company %s does not support QR Code Payments!", rec.company_id.name))
+            if rec.use_qr and (not rec.journal_id.bank_account_id or not rec.journal_id.bank_account_id.proxy_type):
+                raise ValidationError(_("Journal %s needs to have a bank account with correct QR Code fields configured.", rec.journal_id.name))
+
+    @api.depends('use_qr')
+    def _compute_qr(self):
+        for pm in self:
+            pm.default_qr = pm.get_qr_code(False, pm.name, pm.name, pm.company_id.currency_id.id, False)
+
+    def get_qr_code(self, amount, free_communication, structured_communication, currency, debtor_partner,
+                    qr_code_method=None, silent_errors=True, width=200, height=200):
+        """ Generates and returns a QR-code generation URI for this method,
+        raising an error message if something is misconfigured.
+
+        The chosen QR generation method is the one set in qr_method field if there is one,
+        or the first eligible one found. If this search had to be performed and
+        and eligible method was found, qr_method field is set to this method before
+        returning the URL. If no eligible QR method could be found, we return None.
+        """
+        self.ensure_one()
+        if not self.use_qr:
+            return False
+        payment_bank = self.journal_id.bank_account_id
+        debtor_partner = self.env['res.partner'].browse(debtor_partner)
+        currency = self.env['res.currency'].browse(currency)
+        if qr_code_method:
+            # If the user set a qr code generator manually, we check that we can use it
+            error_msg = payment_bank._get_error_messages_for_qr(
+                self.qr_code_method, debtor_partner, currency)
+            if error_msg and not silent_errors:
+                raise UserError(error_msg)
+        else:
+            # Else we find one that's eligible and assign it to the invoice
+            for candidate_method, _candidate_name in self.env['res.partner.bank'].get_available_qr_methods_in_sequence():
+                error_msg = payment_bank._get_error_messages_for_qr(
+                    candidate_method, debtor_partner, currency)
+                if not error_msg:
+                    qr_code_method = candidate_method
+                    break
+
+        if not qr_code_method:
+            # No eligible method could be found; we can't generate the QR-code
+            return None
```

Comment:
I don't think this is needed as you never use the param `qr_code_method`, we can set the `qr_code_method` on the payment method directly. Just add a new field. This code comes from account move which doesn't really make sense in this use case IMO